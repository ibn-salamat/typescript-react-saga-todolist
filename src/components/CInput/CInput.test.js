import { shallow, render, mount } from "enzyme";
import { CInput } from "./index";

const setUp = (props) => shallow(<CInput {...props} />);

describe("Button", () => {
  it("should render component", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });

  it("should call onChange method", () => {
    const event = {
      preventDefault() {},
      target: { value: "value" },
    };

    const mockCallback = jest.fn();
    const component = setUp({ onChange: mockCallback });
    expect(mockCallback.mock.calls.length).toBe(0);

    component.find("input").simulate("change", event);
    expect(mockCallback.mock.calls.length).toBe(1);
  });
});
