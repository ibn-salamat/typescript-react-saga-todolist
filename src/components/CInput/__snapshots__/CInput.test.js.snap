// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render component 1`] = `
<div
  className="c-input"
>
  <input
    onChange={[Function]}
    placeholder="Title"
  />
</div>
`;
